src/code_80027040.c

```diff
@@ -290,39 +111,30 @@ uintptr_t ***gKartTextureTable1[] = {
+//generated by mips_to_c commit 90e635335cb052c4324a3aa7478e4761db19dfaf
 void func_80027040(Player *player, s8 arg1, s8 arg2, s8 arg3, s8 arg4) {
+    struct_D_802DFB80 *temp_s0 = &D_802DFB80[arg4][arg3][arg1];
 
-    //s32 tmp = player->unk_0BC; 
-    void *buf; // = &D_802DFB80[arg4][arg3][arg1 * 0x920];
-    u16 id;
-    if (((((((player->unk_0BC & 0x80) == 0x80) || ((player->unk_0BC & 0x40) == 0x40)) || ((player->unk_0BC & 0x80000) == 0x80000)) || ((player->unk_0BC & 0x800000) == 0x800000)) || ((player->unk_0BC & 0x20000) == 0x20000)) || ((player->unk_044 & 0x800) != 0))
-    {
-        // temp_v1 = player + (arg2 * 2);
+    if (((player->unk_0BC & 0x80) == 0x80) || ((player->unk_0BC & 0x40) == 0x40) || ((player->unk_0BC & 0x80000) == 0x80000) || ((player->unk_0BC & 0x800000) == 0x800000) || ((player->unk_0BC & 0x20000) == 0x20000) || ((player->unk_044 & 0x800) != 0)) {
         if (player->unk_244[arg2] != 0) {
-            //buf = &D_802DFB80[arg2][arg1 * 0x920];
-            buf = &D_802DFB80[arg4][arg3][arg1 * 0x920];
-            osInvalDCache(buf, D_800DDEB0[player->characterId]);
-            id = player->characterId;
-            osPiStartDma(&gDmaIoMesg, 0, 0, &_kart_texturesSegmentRomStart[SEGMENT_OFFSET(gKartTextureTable1[id][player->unk_24C[arg2]][player->unk_244[arg2]])], buf, D_800DDEB0[player->unk_0BC], &gDmaMesgQueue);
+            osInvalDCache(temp_s0, D_800DDEB0[player->characterId]);
+            osPiStartDma(&gDmaIoMesg, 0, 0, &_kart_texturesSegmentRomStart[SEGMENT_OFFSET(gKartTextureTable1[player->characterId][player->unk_24C[arg2]][player->unk_244[arg2]])], temp_s0, D_800DDEB0[player->characterId], &gDmaMesgQueue);
             osRecvMesg(&gDmaMesgQueue, &gMainReceivedMesg, 1);
         } else {
-            buf = &D_802DFB80[arg4][arg3][arg1 * 0x920];
-            osInvalDCache(&D_802DFB80[arg4][arg3], D_800DDEB0[player->characterId]);
-            osPiStartDma(&gDmaIoMesg, 0, 0, &_kart_texturesSegmentRomStart[SEGMENT_OFFSET(gKartTextureTable0[player->characterId][player->unk_24C[arg2]][player->unk_244[arg2]])], buf, D_800DDEB0[player->unk_0BC], &gDmaMesgQueue);
+            osInvalDCache(temp_s0, D_800DDEB0[player->characterId]);
+            osPiStartDma(&gDmaIoMesg, 0, 0, &_kart_texturesSegmentRomStart[SEGMENT_OFFSET(gKartTextureTable0[player->characterId][player->unk_24C[arg2]][player->unk_244[arg2]])], temp_s0, D_800DDEB0[player->characterId], &gDmaMesgQueue);
             osRecvMesg(&gDmaMesgQueue, &gMainReceivedMesg, 1);
         }
-    } else if (((((player->unk_0BC & 0x400) == 0x400) || ((player->unk_0BC & 0x1000000) == 0x1000000)) || ((player->unk_0BC & 0x2000000) == 0x2000000)) || ((player->unk_0BC & 0x10000) == 0x10000))
-    {
-        buf = &D_802DFB80[arg4][arg3][arg1 * 0x920];
-        osInvalDCache(buf, 0x780);
-        osPiStartDma(&gDmaIoMesg, 0, 0, &_kart_texturesSegmentRomStart[SEGMENT_OFFSET(gKartTextureGroup18s[player->characterId][player->unk_0A8 >> 8])], buf, 0x900, &gDmaMesgQueue);
```

Comment:
Definitely a weird section. I was gonna have another look at it but you beat me to it lol.
(I actually mean the section like way below here where your comment is oops)

It's gonna be a few days before I finish reviewing this PR. Mainly because I want to try some things.